### DISCUSSION TITLE
Need help accessing private ESPN Fantasy Basketball league via API

### DISCUSSION THREAD

**Discussion by kzbruh** [cwendt94/espn-api#620] [ESPN API]
Hello! Let me start by saying I'm a complete beginner and this is my first discussion ever. Big thanks to everyone who contributed to the API thus far!

I'm trying to access my private ESPN Fantasy Basketball league using the espn-api library but running into authentication issues. I can access the league via browser, but API requests fail.

**Setup**
League Type: Private H2H 9-category (2024 season)
League URL: https://fantasy.espn.com/basketball/league?leagueId=311923117
Python version: 3.11.11
espn-api version: 0.12.0

**What I've Tried**
Getting fresh cookies after logging in (ESPN_S2, SWID) as well as (ESPN_S2, SWID, DSID)
Using both direct API requests and the espn-api library
Testing different year parameters (2024, 2023)

**Current Implementation**

1. Test Script (src/quick_test.py)

```python
import requests
import toml
from espn_api.basketball import League
import json
from data_fetcher import ESPNDataFetcher

def test_direct_api(secrets: dict, year: int = 2024) -> bool:

    url = (
        f"https://fantasy.espn.com/apis/v3/games/fba/seasons/{year}/segments/0/leagues/"
        f"{secrets['LEAGUE_ID']}"
    )
    cookies = {
        "espn_s2": secrets["ESPN_S2"],
        "SWID": secrets["SWID"],
        "DSID": secrets["DSID"]
    }

    try:
        response = requests.get(url, cookies=cookies)
        print(f"Status code: {response.status_code}")
        return response.status_code == 200
    except Exception as e:
        print(f"Error: {str(e)}")
        return False

def main():
    """
    Main function to test ESPN API access.
    """
    try:
        # Load secrets from the configuration file
        with open('.streamlit/secrets.toml', 'r') as f:
            secrets = toml.load(f)

        # Test direct API access
        direct_api_success = test_direct_api(secrets)
        print(f"Direct API Access: {'✓' if direct_api_success else '✗'}")
    except FileNotFoundError:
        print("Error: '.streamlit/secrets.toml' file not found.")
    except toml.TomlDecodeError:
        print("Error: Unable to decode TOML file.")
    except KeyError as e:
        print(f"Error: Missing key in secrets file: {e}")
    except Exception as e:
        print(f"Unexpected error: {str(e)}")

if __name__ == "__main__":
    main()
```

**Current Results**

```=== ESPN Fantasy Basketball API Test ===
✓ Loaded credentials from secrets.toml

Testing direct API access for year 2024...
Status code: 403
✗ API request failed with status 403
Response: <redirected to ESPN homepage>

Testing ESPN API library access for year 2024...
✗ ESPN API library access failed: League 311923117 does not exist
```

**Environment**
```pandas>=2.1.0
numpy>=2.1.0
streamlit>=1.22.0
requests>=2.28.2
espn_api>=0.12.0
```

Please let me know what am I doing wrong or if I should share anything else to troubleshoot. 

Any help will be much appreciated!

**Comment by cwendt94** [comment]
In your current implementation it doesn't look like you are using the `espn-api` package and directly calling their API. If you want to use the package try updating your code to something like this
``` python3
league = League(league_id=311923117, year=2025, espn_s2=secrets["ESPN_S2"], swid=secrets["SWID"])
```
The package will handle authentication and pull all of your league data
